File: amodeus/upstream/models/room.py
from typing import Annotated

from pydantic import Field

from .base import IDModeusModel
from .building import Building


class Room(IDModeusModel):
    name: str
    name_short: Annotated[str, Field(alias="nameShort")]
    building: Building
    projector_available: Annotated[bool, Field(alias="projectorAvailable")]
    total_capacity: Annotated[int, Field(alias="totalCapacity")]
    working_capacity: Annotated[int, Field(alias="workingCapacity")]

File: src/main/kotlin/me/evgfilim1/amodeus/api/upstream/models/Room.kt
package me.evgfilim1.amodeus.api.upstream.models

import me.evgfilim1.amodeus.api.utils.UUID

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonObject

@Serializable
data class Room(
    override val id: UUID,
    val name: String,
    val nameShort: String,
    val building: Building,
    val projectorAvailable: Boolean,
    val totalCapacity: Int,
    val workingCapacity: Int,
    override val _links: JsonObject,
) : IDModeusModel
